### PR TITLE
[FLINK-1526][gelly] [work in progress] Added MinSpanningTree example, library and test

### PR DESCRIPTION
**Pull request by andralungu**
The purpose of this PR is to receive feedback on how to improve iteratiobns inside Gelly. 
It proves that the current API approach makes it tedious to implement complex algortihms such as the MST. 
There is an ongoing discussion on the dev mailing list which better explains the situation: 
[DISCUSS] Gelly iteration abstractions

### CODE REVIEW

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/MinSpanningTreeExample.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import org.apache.flink.graph.Graph;
+import org.apache.flink.graph.Vertex;
+import org.apache.flink.api.common.ProgramDescription;
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple1;
+import org.apache.flink.api.java.tuple.Tuple3;
+
+public class MinSpanningTreeExample implements ProgramDescription {
```
Comment (hsaputra) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/MinSpanningTreeExample.java]: Could you add Javadoc description of what this class does and the sample output?

We are trying to add contributions to add more comments and descriptions to keep up with new code coming in.

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/utils/MinSpanningTreeData.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+import org.apache.flink.graph.Edge;
+import org.apache.flink.graph.Vertex;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MinSpanningTreeData {
```
Comment (hsaputra) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/utils/MinSpanningTreeData.java]: Could you add Javadoc description of what this class does and relationships to other classes?

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/library/MinSpanningTree.java [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+import org.apache.flink.api.java.tuple.Tuple6;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.graph.example.utils.ConvergenceDetectionAggregator;
+import org.apache.flink.types.BooleanValue;
+import org.apache.flink.util.Collector;
+
+import java.util.Iterator;
+
+@SuppressWarnings("serial")
+public class MinSpanningTree implements GraphAlgorithm<Long, String, Double> {
```
Comment (hsaputra) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/library/MinSpanningTree.java]: Could you add Javadoc description of what this class does and relationships to other classes?

**Comment by vasia** [comment]
Hey @andralungu!

I think we should close this one. We can't really continue from this state anyway. I guess we'll have to revisit this problem once we have for-loop iteration support.

**Pull request closed without merging**